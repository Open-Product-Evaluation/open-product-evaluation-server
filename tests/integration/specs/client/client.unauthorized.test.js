const users = require('../../../seeds/data/user/user')
const clients = require('../../../seeds/data/client/client')
const domains = require('../../../seeds/data/domain/domain')
const { Seeder } = require('mongo-seeding')
const config = require('../../../../config')
const { unauthRequest } = require('../../helper/requests')
const { getSeedID, getNotMatchingID } = require('../../helper/helpers')
const {
  clientAmountQuery,
  createPermanentClientMutation,
  createTemporaryClientMutation,
  loginClientMutation,
  clientsQuery,
  clientQuery,
  updateClientMutation,
  deleteClientMutation,
  setClientOwnerMutation,
  removeClientOwnerMutation
} = require('../../queries/client.queries.js')

const seeder = new Seeder(config.seeder)
const collections = seeder.readCollectionsFromPath(config.seeder.inputPath)

describe('Unauthorized', () => {
  beforeEach(async () => {
    await seeder.import(collections)
  })

  describe('should', async () => {
    it('create a permanent client', async () => {
      const query = createPermanentClientMutation('TestClient', users[0].email)
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(errors).toBeUndefined()
      expect(data.createPermanentClient.client).toMatchSnapshot()
      expect(data.createPermanentClient.code).not.toBeNull()
      expect(data.createPermanentClient.token).not.toBeNull()
    })
    it('login as a permanent client', async () => {
      const query = loginClientMutation(users[0].email, clients[2].code)
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(errors).toBeUndefined()
      expect(data.loginClient.client).toMatchSnapshot()
      expect(data.loginClient.code).not.toBeNull()
      expect(data.loginClient.token).not.toBeNull()
    })
    it('create a temporary client', async () => {
      const query = createTemporaryClientMutation(getSeedID(domains[0]))
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(errors).toBeUndefined()
      expect(data.createTemporaryClient.client).toMatchSnapshot()
      expect(data.createTemporaryClient.token).not.toBeNull()
    })
  })
  describe('should not', async () => {
    it('get the amount of clients', async () => {
      const query = clientAmountQuery()
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(data.clientAmount).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('find any when querying for clients', async () => {
      const query = clientsQuery()
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(data.clients).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('find it when querying for a specific client', async () => {
      const query = clientQuery(getSeedID(clients[0]))
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('create a permanent Client with an email that does not belong to a user', async () => {
      const query = createPermanentClientMutation('TestClient', 'noUser@found.com')
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('login as a permanent Client with an email that does not belong to a owner', async () => {
      const query = loginClientMutation('notMy@owner.com', clients[2].code)
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('login as a permanent Client with an invalid code', async () => {
      const query = loginClientMutation(users[0].email, 'invalidCode')
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('create a temporary Client with an ID that does not belong to a domain', async () => {
      const query = createTemporaryClientMutation(getNotMatchingID())
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('update a client', async () => {
      const query = updateClientMutation(getSeedID(clients[0]),
        { name: 'notAllowedNameChange' },
        true
      )

      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('delete a client', async () => {
      const query = deleteClientMutation(getSeedID(clients[0]))
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('set an owner for a client', async () => {
      const query = setClientOwnerMutation(getSeedID(clients[0]), users[0].email)
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('remove an owner of a client', async () => {
      const query = removeClientOwnerMutation(getSeedID(clients[0]), getSeedID(users[1]))
      const res = await unauthRequest(query)
      const { data, errors } = res

      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
  })
})
