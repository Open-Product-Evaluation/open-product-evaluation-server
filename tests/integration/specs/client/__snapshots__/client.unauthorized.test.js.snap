// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unauthorized should create a permanent client 1`] = `
Object {
  "name": "TestClient",
}
`;

exports[`Unauthorized should create a temporary client 1`] = `
Object {
  "domain": Object {
    "activeQuestion": null,
    "activeSurvey": Object {
      "id": "f7508d9bce487edf338f3defbecfe7dcda3091d24fa9676580aec77771600de3",
      "questions": Array [
        Object {
          "id": "1bb94534c8e18ea82a7cb685edb89b018a14792f36797faa7c3e3bc877b38a44",
          "value": "Sind Sie Student der TH Köln?",
        },
        Object {
          "id": "0cbb7f5ba03e09d8fd3316b1772d66ff58c081bb9c1850990cff4bfac9a23bc3",
          "value": "Sind Sie zufrieden mit der Mensa am Campus Gummersbach?",
        },
        Object {
          "id": "6150c997e679eba9aaeb6e44d8107e43f45ef91c37a150c2fdc00525f3e62a72",
          "value": "Wie oft sind Sie durchschnittlich pro Woche in der Hochschule?",
        },
        Object {
          "id": "a7eba9cd6d6fb2db96d1b017f233e64a07b7bca182368194081cd94525bcd4d6",
          "value": "Wie zufrieden sind Sie mit dem Essen in der Mensa?",
        },
        Object {
          "id": "193e16a4088aa2a179e4323ae1efe99cfa166259576c45f548e5d46ea017df02",
          "value": "Welchen Professor finden Sie am kompetentesten?",
        },
        Object {
          "id": "94e77c576c68aa3adbba713fb341d68283d984853fbe15a07940c9a4d8988b16",
          "value": "Sortieren Sie diese Fächer nach dem subjektiven Lerngewinn.",
        },
      ],
      "title": "Umfrage zur Zufriedenheit an der Technischen Hochschule Köln",
    },
    "id": "24fbef976227ca6b503a99e795ba4812656b3f97ff6cd7c291b450e1cbf2d522",
    "name": "TH Eingangshalle",
  },
  "name": "Temporary Client",
}
`;

exports[`Unauthorized should login as a permanent client 1`] = `
Object {
  "code": "P3DRU72GQ",
  "domain": Object {
    "activeQuestion": null,
    "activeSurvey": Object {
      "id": "c9f789ae90d2a491f15794da7c7de88cd4f9fc267eee0112fa4cc947d25f7ae3",
      "questions": Array [
        Object {
          "id": "f0a5499c989ef2579982d362d8c374ec3c6f9f3ec34ee0a9204b58b26f616219",
          "value": "Wie gefallen Ihnen diese Kirschen?",
        },
        Object {
          "id": "5f98baec8f2b489c8e87eab79f7adf490f54e3b08f7878599d77164124ad1795",
          "value": "Essen Sie gerne Obst?",
        },
        Object {
          "id": "544920d8dc19d59ca0fdeaeff0f127935d490d31e6c9147223489168d963bf8f",
          "value": "Wie viel geben Sie für Obst in der Woche aus?",
        },
        Object {
          "id": "495103000d697a14339dfa1ad261e234ff885239e9e3b4b95a7d5f45e9d90cb6",
          "value": "Wie viel Obst essen Sie am Tag?",
        },
        Object {
          "id": "979eb275a748639d3d57f563cf2879c836f4166cd323fc42ce3e0f81936ed192",
          "value": "Was ist ihr Lieblingsobst?",
        },
        Object {
          "id": "1b6cdfb57ded968d69ffe801a6b184118861bce82906cce0c0b9e53f40e8cd8e",
          "value": "Welches Obst essen Sie am häufigsten?",
        },
      ],
      "title": "Umfrage zum Konsumverhalten von Obst",
    },
    "id": "495b1a7b6cfca552428759780117761c164310b6557c3f06e4357fafc8cb4453",
    "name": "Kölner Markt",
  },
  "id": "61b9c12fb4409fda98fa7ab11e0d13e94a4fce60330a7838cecf5700abdc51f3",
  "name": "Markttablet",
}
`;
